meteor/client/ui/Shelf/MiniRundownPanel.tsx

```diff
@@ -0,0 +1,241 @@
+					<span className="mini-rundown-panel__rank">{next.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{next.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{next.partName}
+					</span>
+				</div>
+
+				<div className="second-part">
+					<span className="mini-rundown-panel__rank">{second.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{second.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{second.partName}
+					</span>
+				</div>
+
+				<div className="third-part">
+					<span className="mini-rundown-panel__rank">{third.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{third.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{third.partName}
+					</span>
+				</div>
+			</div>
+		)
+	}
+}
+
+export const MiniRundownPanel = withTracker<IMiniRundownPanelProps, IState, IMiniRundownPanelTrackedProps>(
+	(props: IMiniRundownPanelProps & IMiniRundownPanelTrackedProps) => {
+		let currentPartInstance: PartInstance | undefined = undefined
+		let currentSegment: Segment | undefined = undefined
+		let nextPartInstance: PartInstance | undefined = undefined
+		let nextSegment: Segment | undefined = undefined
+
+		const currentPartInstanceId: PartInstanceId | null = props.playlist.currentPartInstanceId
+		if (currentPartInstanceId) {
+			currentPartInstance = PartInstances.findOne(currentPartInstanceId)
+		}
+
+		if (currentPartInstance) {
+			currentSegment = Segments.findOne(currentPartInstance.segmentId)
+		}
+
+		if (props.playlist.nextPartInstanceId) {
+			nextPartInstance = PartInstances.findOne(props.playlist.nextPartInstanceId)
+		}
+
+		if (nextPartInstance) {
+			nextSegment = Segments.findOne(nextPartInstance.segmentId)
+		}
+		return { currentSegment, currentPartInstance, nextPartInstance, nextSegment }
+	},
+	(_data, props: IMiniRundownPanelProps, nextProps: IMiniRundownPanelProps) => {
+		return !_.isEqual(props, nextProps)
+	}
+)(MiniRundownPanelInner)
+
+function getContainerClass(props, isDashboardLayout: boolean): string[] | undefined {
+	return isDashboardLayout ? (props.panel as DashboardLayoutMiniRundown).customClasses : undefined
+}
+
+function getContainerStyle(props, isDashboardLayout: boolean): any {
+	return _.extend(isDashboardLayout ? dashboardElementPosition({ ...(props.panel as DashboardLayoutNextInfo) }) : {}, {
+		visibility: props.visible ? 'visible' : 'hidden',
+	})
+}
+
+function getElementStyle(props, isDashboardLayout: boolean) {
+	return {
+		fontSize: isDashboardLayout ? ((props.panel as DashboardLayoutMiniRundown).scale || 1) + 'em' : undefined,
+	}
+}
+
+function isShowAnyValues(props): boolean {
+	return !props.panel.hideForDynamicallyInsertedParts || props.nextPartInstance?.orphaned !== 'adlib-part'
+}
```

Comment:
I'm not entirely sure if `hideForDynamicallyInsertedParts` works as expected here. Do I understand it right that it would essentially hide all the contents when enabled, and the current part is dynamic?

Reply:
No it doesn't. Does it even make sense for this feature?